### PR TITLE
Http2 keywords 60x v5

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/4067

Describe changes:
- Allows some HTTP1 keywords one HTTP2 traffic

A configuration option allows to have all HTTP1 rules working on HTTP2 traffic without specifying it for each rule

Modifies https://github.com/OISF/suricata/pull/6141 with 
- allowing `alert http` and `alert ip` to work on HTTP2 traffic

With Signature alproto and Flow alproto, the options were 
- have only one alproto for HTTP1 and HTTP2, but this needed a major rework (as I see it)
- Duplicate signatures so that a HTTP1 signature has also a HTTP2 version, as is done with bidirectional signatures, but that linked list code has been a trap for bugs, so I avoided touching it
- Have a signature alproto match multiple Flow alproto : That is the solution I chose as in master. The difference with master is that we do not have HTTP_ANY, just a config option to tell that all HTTP1 rules should try to match on HTTP2 traffic

suricata-verify-pr: 503

Makes S-V test in OISF/suricata-verify#503 pass

The S-V tests `http2-keywords` and `http2-keywords2` still require version 7 because they have rules with `alert http1` which is not recognized here.
What could we do about it ?
- have a signature keyword to load the rule only if the suricata version is good for it ?
- allow to load some rules even if some other rules error ?

### CODE REVIEW

**Comment by catenacyber** [comment]
Also fixed the commit-check failure by squashing 2 commits together

**Comment by catenacyber** [comment]
CIFuzz failure should be fixed by #6140

**Comment by catenacyber** [comment]
Needs #6143 to get the failing builds to work

File: src/app-layer-parser.c [hunk 1]
```diff
@@ -215,10 +215,16 @@ void AppLayerParserStateFree(AppLayerParserState *pstate)
...
 }
 
+bool g_config_http1keywords_http2traffic = false;
+
 int AppLayerParserSetup(void)
 {
     SCEnter();
     memset(&alp_ctx, 0, sizeof(alp_ctx));
+    int value = 0;
+    if (ConfGetBool("app-layer.protocols.http.rules-on-http2", &value) == 1 && value == 1) {
```
Comment (victorjulien) [inline src/app-layer-parser.c:225]: I think it would make more sense to make this part of `app-layer.protocols.http2`.

Comment (catenacyber) [reply src/app-layer-parser.c:225]: indeed, will do

**Comment by catenacyber** [comment]
Replaced by https://github.com/OISF/suricata/pull/6152

**Pull request closed without merging**